src/dscom/RegistrationServices.cs

```diff
@@ -283,6 +283,18 @@ private static IReadOnlyCollection<Type> GetComRegistratableTypes(Assembly assem
     {
         static bool TypeMustBeRegistered(Type type)
         {
+            // Only public types are com visible
+            if (!type.IsPublic && !type.IsNestedPublic && !type.IsByRef)
+            {
+                return false;
+            }
+
+            // Type could be defined as not com visible
+            if (!type.IsComVisible())
```

Comment:
Isn't that solved by Marshal.IsTypeVisibleFromCom(type); below?

Reply:
It does not respect the ComVisibleAttribute with 'False'

Reply:
the check Marshal.IsTypeVisibleFromCom(type) will not be applied, if it is a class or valuetype.
https://github.com/dspace-group/dscom/blob/530e7fbe78e47c6925fb959a2eb7fc5b77dcdf15/src/dscom/RegistrationServices.cs#L302-L307